examples/access-log-policy-example.yaml

```diff
@@ -0,0 +1,10 @@
+apiVersion: application-networking.k8s.aws/v1alpha1
+kind: AccessLogPolicy
+metadata:
+  name: test-policy
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
+    name: my-hotel
+  destinationArn: "arn:aws:s3:::my-bucket"
```

Comment:
Just a general question on the feature. How are we planning on reporting back the actual log paths? From what I recall, logs are based on unique resource ids, not names. How will folks using the controller know which logs apply to which k8s resources?

Reply:
Not 100% sure I follow, so please correct me if I'm going down the wrong path here. When a user creates a Gateway or Route, that resource is associated to 1 VPC Lattice Service Network or Service. When they create an AccessLogPolicy with a targetRef of a Gateway or Route, it's the equivalent of creating a VPC Lattice Access Log Subscription for the corresponding ServiceNetwork or Service.

So if I create Gateway `my-hotel`, which in VPC Lattice is Service Network `sn-12345678901234567`, and then create an AccessLogPolicy which refers to Gateway `my-hotel`, then it will create an AccessLogSubscription with resourceIdentifier `sn-12345678901234567`, and any access logs flowing through that Service Network will wind up in the log destination I specified in the AccessLogPolicy.

Reply:
Right. So the path in S3 would look like [[1](https://docs.aws.amazon.com/vpc-lattice/latest/ug/monitoring-access-logs.html)]:

`[bucket]/[prefix]/AWSLogs/[accountId]/VpcLattice/AccessLogs/[region]/[YYYY/MM/DD]/[resource-id]/[accountId]_VpcLatticeAccessLogs_[region]_[resource-id]_YYYYMMDDTHHmmZ_[hash].json.gz
`

Let's assume I have a number of gateways I'm managing - how will I know the Lattice resource ID for each? For example, do we patch the gateway with the Lattice service network ID? I see it as a message here: https://github.com/aws/aws-application-networking-k8s/blob/main/controllers/gateway_controller.go#L279. Ideally, I would be able to find out the ID by doing something like ```kubectl get gateway```.

Similarly, it may be difficult to map from a k8s route to a log line, or from a log line back to the k8s objects, since these log entries also use unique IDs rather than k8s-based names. Maybe for now just enabling the logs is enough, then customers can use the Lattice API out of band to determine the ID. Longer term we should look a improving this experience.

Reply:
Agreed, this would absolutely improve clarity for users. I'll be sure to include the ALS id in the AccessLogPolicy, and we can create an issue to add this behaviour to other resources.